nomad/structs/structs.go

```diff
@@ -435,6 +447,30 @@ func (w WriteRequest) AllowStaleRead() bool {
 	return false
 }
 
+func (w *WriteRequest) SetIdentity(identity *AuthenticatedIdentity) {
+	w.identity = identity
+}
+
+func (w WriteRequest) GetIdentity() *AuthenticatedIdentity {
+	return w.identity
+}
+
+type AuthenticatedIdentity struct {
```

Comment:
docstring at a least enough to hint at folks where to learn more about this abstraction

Especially the fact that the fields are largely used independently makes its actual use different from what someone might assume at a glance.